examples/pytorch/model_zoo/chem/generative_models/dgmg/README.md

```diff
@@ -0,0 +1,107 @@
+
+### Preprocessing
+
+With our implementation, this model has several limitations:
+1. Information about protonation and chirality are ignored during generation
+2. Molecules consisting of `[N+]`, `[O-]`, etc. cannot be generated.
+
+For example, the model can only generate `O=C1NC(=S)NC(=O)C1=CNC1=CC=C(N(=O)O)C=C1O` from 
+`O=C1NC(=S)NC(=O)C1=CNC1=CC=C([N+](=O)[O-])C=C1O` even with the correct decisions.
+
+To avoid issues about validity and novelty, we filter out these molecules from the dataset.
+
+### ChEMBL
+
+The authors use the [ChEMBL database](https://www.ebi.ac.uk/chembl/). Since they 
+did not release the code, we use a subset from [Olivecrona et al.](https://github.com/MarcusOlivecrona/REINVENT), 
+another work on generative modeling. 
+
+The authors restrict their dataset to molecules with at most 20 heavy atoms, and used a training/validation
+split of 130, 830/26, 166 examples each. We use the same split but need to relax 20 to 23 as we are using
+a different subset.
+
+### ZINC
+
+After the pre-processing, we are left with 232464 molecules for training and 5000 molecules for validation.
+
+## Usage
+
+### Training
+
+Training auto-regressive generative models tends to be very slow. According to the authors, they use multiprocess to
+speed up training and gpu does not give much speed advantage. We follow their approach and perform multiprocess cpu
+training.
+
+To start training, do `python train.py -np X -d Y -o Z`, where:
+- `X` is the number of processes to use. We use `32`. 
+- `Y` is the dataset to use. We currently support `ChEMBL` and `ZINC`. 
+- `Z` is the order for generating decision sequences, which can be either `canonical`, or `random`.
+
+Even though multiprocess yields a significant speedup comparing to a single process, the training can still take a long 
+time (several days). An epoch of training and validation can take up to one hour and a half on our machine. If not necessary, we 
+recommend users use our pre-trained models. 
+
+Meanwhile, we make a checkpoint of our model whenever there is a performance improvement on the validation set so you 
+do not need to wait until the training terminates.
+
+#### Dataset configuration
+
+You can also use your own dataset with `python train.py -np X -d Y -o Z -tf A -vf B`, where:
+- `A` is the path to a file for the training data with one SMILES a line
+- `B` is the path to a file for the validation data with one SMILES a line
+
+#### Monitoring
+
+We can monitor the training process with tensorboard as below:
+
+![](https://s3.us-east-2.amazonaws.com/dgl.ai/model_zoo/drug_discovery/dgmg/tensorboard.png)
+
+To use tensorboard, you need to install [tensorboardX](https://github.com/lanpa/tensorboardX) and 
+[TensorFlow](https://www.tensorflow.org/). You can lunch tensorboard with `tensorboard --logdir=.`
+
+### Evaluation
+
+To evaluate your own model, do `python eval.py -d X -o Y -p Z -np A -mn B -gt C`, where:
+- `X` is the dataset to use, such as `ChEMBL` and `ZINC`
```

Comment:
```
optional arguments:
  -h, --help            show this help message and exit
  -s SEED, --seed SEED  random seed
  -w WARMUP_EPOCHS, --warmup-epochs WARMUP_EPOCHS
                        Number of epochs where no lr decay is performed.
  -d DATASET, --dataset DATASET
                        dataset to use
  -o {random,canonical}, --order {random,canonical}
                        order to generate graphs
  -tf TRAIN_FILE, --train-file TRAIN_FILE
                        Path to a file with one SMILES a line for training
                        data. This is only necessary if you want to use a new
                        dataset.
  -vf VAL_FILE, --val-file VAL_FILE
                        Path to a file with one SMILES a line for validation
                        data. This is only necessary if you want to use a new
                        dataset.
  -l LOG_DIR, --log-dir LOG_DIR
                        folder to save info like experiment configuration
  -np NUM_PROCESSES, --num-processes NUM_PROCESSES
                        number of processes to use
```

XYZABC is hard to track when reading for me. `python train.py --help` generate acceptable result shows above. You can merge your description to python file. And I found [this](https://stackoverflow.com/questions/12151306/argparse-way-to-include-default-values-in-help) could add default value to `--help` also